src/main/java/org/candlepin/subscriptions/resource/AbstractReportResource.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2009 - 2019 Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource;
+
+import org.candlepin.subscriptions.exception.ErrorCode;
+import org.candlepin.subscriptions.exception.SubscriptionsException;
+
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.core.Response;
+
+/**
+ * Functionality common to both capacity and tally resources.
+ */
+public class AbstractReportResource {
+
+    private static final Integer DEFAULT_LIMIT = 50;
+
+    protected String getAccountNumber() {
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+        return auth.getName();
+    }
+
+    @NotNull
+    Pageable getPageable(Integer offset, Integer limit) {
+        if (limit == null) {
+            limit = DEFAULT_LIMIT;
```

Comment:
Looking later on in the code, it appears that this method is meant to cover the bases if offset has been specified alone, if limit has been specified alone, and if they have both been specified.

I'd like to suggest another way.  We consider it a bad request if only one of the offset and limit have been given (and handle this via the validation level).  Optionally, for offset we can put a [default value](https://swagger.io/docs/specification/describing-parameters/#default) of zero in the API definition which results in a JAX-RS `@DefaultValue` being inserted into the method signature.  In that case, only the presence of a limit would be diagnostic of a request for paging.

We can then do the second part (where we check that the offset is a multiple of the limit) as a custom validation.  This approach removes the need for both resource classes to inherit from the same class and moves the concerns (setting a default and validation) out of the method itself (where someone might remove it or it gets clobbered in a merge) and into a higher level.

The `getAccountNumber` method can either be moved to a static utility class or just removed since it's not all that complicated.

Also, we need to validate that limit is non-zero (probably we want it to be positive) if we're going to be dividing by it at any point.

Reply:
> Looking later on in the code, it appears that this method is meant to cover the bases if offset has been specified alone, if limit has been specified alone, and if they have both been specified.

> I'd like to suggest another way. We consider it a bad request if only one of the offset and limit have been given (and handle this via the validation level). Optionally, for offset we can put a default value of zero in the API definition which results in a JAX-RS @DefaultValue being inserted into the method signature. In that case, only the presence of a limit would be diagnostic of a request for paging.

Unfortunately we have to perform paging if either is specified (i.e. is valid to specify offset=10 without specifying limit), so we still have to check for the presence of either, so I'm inclined to not change it. However, I'm happy to revisit whether we decouple the offset/limit validation into another layer at some point.

> The getAccountNumber method can either be moved to a static utility class or just removed since it's not all that complicated.

I changed AsbtractReportResource to ResourceUtils by making methods static.

> Also, we need to validate that limit is non-zero (probably we want it to be positive) if we're going to be dividing by it at any point.

I added `minimum: 1` to the spec for this.